product_brand/models/product_brand.py

```diff
@@ -31,5 +31,13 @@ class ProductBrand(models.Model):
 
     @api.depends("product_ids")
     def _compute_products_count(self):
+        product_model = self.env["product.template"]
+        groups = product_model.read_group(
+            [("product_brand_id", "in", self.ids)],
+            ["product_brand_id"],
+            ["product_brand_id"],
+            lazy=False,
```

Comment:
As information, no need of this, as it's the same lazy or not when you only have one grouping column.